src/vs/workbench/contrib/debug/browser/statusbarColorProvider.ts

```diff
@@ -61,15 +62,25 @@ export class StatusBarColorProvider implements IWorkbenchContribution {
 	constructor(
 		@IDebugService private readonly debugService: IDebugService,
 		@IWorkspaceContextService private readonly contextService: IWorkspaceContextService,
-		@IStatusbarService private readonly statusbarService: IStatusbarService
+		@IStatusbarService private readonly statusbarService: IStatusbarService,
+		@IConfigurationService private readonly configurationService: IConfigurationService
 	) {
 		this.debugService.onDidChangeState(this.update, this, this.disposables);
 		this.contextService.onDidChangeWorkbenchState(this.update, this, this.disposables);
+		this.configurationService.onDidChangeConfiguration((e) => {
+			if (e.affectsConfiguration('debug.enableStatusBarColor')) {
+				this.update();
+			}
+		});
 		this.update();
 	}
 
 	protected update(): void {
 		this.enabled = isStatusbarInDebugMode(this.debugService.state, this.debugService.getModel().getSessions());
```

Comment:
Can you write this method to only set `this.enabled` once per call?

Reply:
the update() method?

Reply:
yeah, fixed